src/middlewared/middlewared/plugins/jail.py

```diff
@@ -117,6 +117,168 @@ def query(self, filters=None, options=None):
 
         return filter_list(jails, filters, options)
 
+    @accepts()
```

Comment:
It almost seems like we need a new service called "plugin" to separate the logic?

Could be in the same file.

Thoughts?

Reply:
@william-gr i couldn't agree more. I was afraid of breaking the API too much this way. `fetch` too is overloaded and having a new service will help us manage that too in a better way. Please advise if you think i should move ahead with this and update the PR.

Reply:
Yes, lets go ahead. We can try to somehow keep backward compatibility if at all possible, but currently there is too much confusion in the API with jails vs plugins